tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -806,23 +786,21 @@ public String registerWorkerSpec(String taskId, String spec) throws IOException
       }
     }
 
+    /**
+     * This method was used to workaround an issue that the Python script finished while the container failed to
+     * close due to GPU allocation issue, which doesn't exist anymore.
+     *
+     * Discussion: A benefit of decoupling registering execution result from container exit status is that we can decouple
+     * tony from a specific resource manager's callback logic.
+     * However, this easily introduces lots of bugs since we'd have 3 places to handle failures - register call, container
+     * complete callback and heartbeat.
+     * To make things easier, we decide to go back and piggyback on container completion to dictate the execution result
+     * of a task.
+     * Still keep this RPC call here for now.
```

Comment:
Can you add a TODO here and file an Issue to remove this method if we plan to just rely on container exit status going forward?

Reply:
Not really want to remove this method as this still provides some debugging information to see if it is a non-zero exit code from script or from container.